### PR TITLE
feat: enable data_cache_config for postgresql read replicas

### PR DESCRIPTION
**Pull request by sofuture**
Read replicas for Enterprise Plus Postgres instances can have data caching enabled. This change should allow it to be configured correctly.

### CODE REVIEW

File: modules/postgresql/read_replica.tf [hunk 1]
```diff
@@ -104,6 +104,12 @@ resource "google_sql_database_instance" "replicas" {
       zone = lookup(each.value, "zone", local.zone)
     }
 
+    dynamic "data_cache_config" {
+      for_each = var.edition == "ENTERPRISE_PLUS" && var.data_cache_enabled ? ["cache_enabled"] : []
+      content {
+        data_cache_enabled = var.data_cache_enabled
+      }
+    }
```
Comment (ChristopheMaingard) [inline modules/postgresql/read_replica.tf:112]: How about doing like this ?

```suggestion
    dynamic "data_cache_config" {
      for_each = lookup(each.value, "edition", var.edition) == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled)  ? ["cache_enabled"] : []
      content {
        data_cache_enabled = lookup(each.value, "data_cache_enabled", var.data_cache_enabled)
      }
    }

```

It just requires to add  `data_cache_enabled = optional(bool)` in the variable `read_replicas`.
That way the configuration could be more granular :)

Comment (sofuture) [reply modules/postgresql/read_replica.tf:112]: That's a good suggestion, thanks!

**Comment by imrannayer** [comment]
Fied issues in #620. Will test this PR after merging #620.

File: modules/postgresql/read_replica.tf (hunk 1 above)
Comment (ChristopheMaingard) [reply modules/postgresql/read_replica.tf]: Thanks!
You just need to add the option within variables.tf
```
variable "read_replicas" {
  description = "List of read replicas to create. Encryption key is required for replica in different region. For replica in same region as master set encryption_key_name = null"
  type = list(object({
    name                  = string
    name_override         = optional(string)
     ....
    data_cache_enabled = optional(bool)
  }))
  default = []
}
```

**Comment by imrannayer** [comment]
@sofuture integration test failing:

```
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185: Error: Operation failed
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185: 
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:   on ../../modules/postgresql/read_replica.tf line 108, in resource "google_sql_database_instance" "replicas":
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:  108:       for_each = lookup(each.value, "edition", var.edition) == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled) ? ["cache_enabled"] : []
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:     ├────────────────
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:     │ each.value is object with 16 attributes
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:     │ var.data_cache_enabled is false
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185:     │ var.edition is null
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185: 
TestPostgreSqlPscModule 2024-07-23T08:57:52Z command.go:185: Error during operation: argument must not be null.
TestPostgreSqlPscModule 2024-07-23T08:57:52Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
Error: Operation failed

  on ../../modules/postgresql/read_replica.tf line 108, in resource "google_sql_database_instance" "replicas":
 108:       for_each = lookup(each.value, "edition", var.edition) == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled) ? ["cache_enabled"] : []
    ├────────────────
    │ each.value is object with 16 attributes
    │ var.data_cache_enabled is false
    │ var.edition is null

Error during operation: argument must not be null.}
```

**Comment by sofuture** [comment]
@imrannayer can `edition` be set to default to `ENTERPRISE` instead of null? As I understand it is a binary value in practice, and not actually nullable.

File: modules/postgresql/read_replica.tf [hunk 2]
```diff
@@ -104,6 +104,13 @@ resource "google_sql_database_instance" "replicas" {
       zone = lookup(each.value, "zone", local.zone)
     }
 
+    dynamic "data_cache_config" {
+      for_each = lookup(each.value, "edition", var.edition) == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled) ? ["cache_enabled"] : []
```
Comment (imrannayer) [inline modules/postgresql/read_replica.tf:108]: can u try following code?

```
coalesce(each.value.edition, var.edition, "ENTERPRISE") == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled) ? ["cache_enabled"] : []
```

**Review by imrannayer — APPROVED** [review]

**Pull request merged**